cloud/libvirt/actuators/machine/actuator.go

```diff
@@ -112,8 +112,12 @@ func (a *Actuator) Delete(cluster *clusterv1.Cluster, machine *clusterv1.Machine
 	}
 
 	defer client.Close()
-
-	return deleteVolumeAndDomain(machine, client)
+	exists, err := libvirtutils.DomainExists(machine.Name, client)
+	if exists {
+		return deleteVolumeAndDomain(machine, client)
+	} else {
```

Comment:
Drop the `else`.

Reply:
also can we add a log saying something like "skipping deletion, machine does not exist" or so